app/src/sharedTest/java/org/oppia/app/testing/DragDropTestActivityTest.kt

```diff
@@ -0,0 +1,87 @@
+package org.oppia.app.testing
+
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.GeneralLocation
+import androidx.test.espresso.action.Press
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPosition
+import org.oppia.app.utility.ChildViewCoordinatesProvider
+import org.oppia.app.utility.CustomGeneralLocation
+import org.oppia.app.utility.DragViewAction
+import org.oppia.app.utility.RecyclerViewCoordinatesProvider
+
+@RunWith(AndroidJUnit4::class)
+class DragDropTestActivityTest {
```

Comment:
This is a bit confusing to me. I don't think we need to test a test activity. We want to test a production-facing class or a test utility. I'd expect the test to be specific to the drag & drop callback class, and that it would include arrangement code for setting up the callback class in some sort of test harness (like a test activity).

Reply:
Actually i did a testing activity because this helper class. Isn't attached to any production facing code as of in the PR also drag drop callback won't work without an activity/recycler view so we need a activity so that we can attach it to the view.